src/hooks/useCampPartygoers.ts

```diff
@@ -1,30 +1,49 @@
-import { useMemo } from "react";
+import { useCallback, useMemo } from "react";
 
 import { User } from "types/User";
 import { WithId } from "utils/id";
 import { partygoersSelector } from "utils/selectors";
 
 import { useConnectPartyGoers } from "./useConnectPartyGoers";
+import { useConnectRelatedVenues } from "./useConnectRelatedVenues";
 import { useSelector } from "./useSelector";
 import { useUserLastSeenThreshold } from "./useUserLastSeenThreshold";
+import { useVenueId } from "./useVenueId";
 
 /**
  * Hook to retrieve a list of partygoers who were in venueName
  * within lastSeenThreshold.
  *
  * @param venueName venueName to filter partygoers by
  */
-export const useCampPartygoers = (venueName: string): WithId<User>[] => {
+export const usePartygoersThreshold = (venueName: string): WithId<User>[] => {
   useConnectPartyGoers();
+  const venueId = useVenueId();
   const partygoers = useSelector(partygoersSelector) ?? [];
 
+  const { parentVenue, relatedVenues } = useConnectRelatedVenues({
+    venueId,
+    withEvents: true,
+  });
+
   const lastSeenThreshold = useUserLastSeenThreshold();
 
-  return useMemo(
-    () =>
-      partygoers.filter(
-        (partygoer) => partygoer?.lastSeenIn?.[venueName] > lastSeenThreshold
-      ),
-    [partygoers, lastSeenThreshold, venueName]
+  const parentVenueName = parentVenue?.name ?? "";
```

Comment:
Instead of defaulting with `?? ""` here, would it be better to use a 'guard' to bail out early when `parentVenueName` is `undefined`? And not try to filter by it or similar?

Reply:
Sounds like a reasonable thing to do. Will add guard.